sites/public/lib/hooks.ts

```diff
@@ -30,3 +32,36 @@ export const useFormConductor = (stepName: string) => {
   }, [conductor])
   return context
 }
+
+const listingsFetcher = function (listingsService: ListingsService) {
+  return (_url: string, page: number, limit: number) => {
+    const params = {
+      page,
+      limit,
+    }
+    return listingsService.list(params)
+  }
+}
+
+// TODO: move this so it can be shared with the partner site.
+export function useListingsData(pageIndex: number, limit = 10) {
+  const { listingsService } = useContext(ApiClientContext)
+  const { data, error } = useSWR(
+    [`${process.env.backendApiBase}/listings`, pageIndex, limit],
+    listingsFetcher(listingsService)
+  )
+
+  return {
+    listingsData: data,
+    listingsLoading: !error && !data,
+    listingsError: error,
+  }
+}
+
+export const usePrevPage = (page: number) => {
```

Comment:
Please add a comment about what this does.

Reply:
Done, hopefully this is more useful for filters.